app/scripts/controllers/detect-tokens.js

```diff
@@ -101,15 +96,6 @@ export default class DetectTokensController extends StaticIntervalPollingControl
       }
     });
 
-    tokensController?.subscribe(
-      ({ tokens = [], ignoredTokens = [], detectedTokens = [] }) => {
-        this.tokenAddresses = tokens.map((token) => {
-          return token.address;
-        });
-        this.hiddenTokens = ignoredTokens;
-        this.detectedTokens = detectedTokens;
-      },
-    );
```

Comment:
`tokensController?.subscribe` was typically the last event to fire during a chain switch.  Therefore `tokenAddresses`, `hiddenTokens`, and `detectedTokens` were still from the old chain during detection, even when (most) everything else is pointing to the new chain.  Since `tokensController` has state per-chain, we can just grab the tokens for the correct chain instead of waiting to be updated.